api/bin/chainflip-broker-api/src/main.rs

```diff
@@ -323,10 +326,28 @@ impl RpcServer for RpcServerImpl {
 
 	async fn register_affiliate(
 		&self,
-		short_id: AffiliateShortId,
 		affiliate_id: AccountId32,
-	) -> RpcResult<H256> {
-		Ok(self.api.broker_api().register_affiliate(short_id, affiliate_id).await?)
+		short_id: Option<AffiliateShortId>,
+	) -> RpcResult<AffiliateShortId> {
+		self.api.broker_api().register_affiliate(affiliate_id.clone(), short_id).await?;
+
+		// Check that the affiliate was registered and get its short id
+		let affiliates = self.api.query_api().get_affiliates(None, None).await?;
+		if let Some((assigned_short_id, _)) = affiliates.iter().find(|(_, id)| *id == affiliate_id)
+		{
+			if let Some(given_short_id) = short_id {
+				if given_short_id != *assigned_short_id {
+					return Err(anyhow!("Failed to update affiliate registration").into());
+				}
+			}
+			return Ok(*assigned_short_id).map_err(BrokerApiError::Other);
```

Comment:
Why do it like this instead of using the register_affiliate extrinsic return value / events?